src/rust/engine/src/externs/process.rs

```diff
@@ -29,17 +29,22 @@ impl PyProcessExecutionEnvironment {
         platform: String,
         remote_execution: bool,
         remote_execution_extra_platform_properties: Vec<(String, String)>,
+        execute_in_workspace: bool,
         environment_name: Option<String>,
         docker_image: Option<String>,
     ) -> PyResult<Self> {
```

Comment:
Oy. Need support for ADTs at this boundary. Oh well!

Reply:
> Oy. Need support for ADTs at this boundary. Oh well!

What's the preferred way to do that? A Python class per enum variant with them all in a `Union`?

Reply:
Also, I will explore such a refactor after this PR has landed.

Reply:
Hm... haven't thought about it. It's not a pattern that we've used yet I think.

But. Probably you would do it by defining a Python class on the Rust side that was constructed with one of a series of factory functions. Internally it would just have a Rust `enum`.

Reply:
I was thinking of separate Python classes for each enum variant since it allows mypy to check whether all types in a  union (e.g., `Variant1 | Variant2 | Variant3`) have been handled. Prior PyO3 discussion is here: https://github.com/PyO3/pyo3/issues/417 (although seems still like no consensus)